common/customizable_schema.go

```diff
@@ -30,6 +90,16 @@ func CustomizeSchemaPath(s map[string]*schema.Schema, path ...string) *Customiza
 	return &CustomizableSchema{Schema: sch, path: path}
 }
 
+func (s *CustomizableSchema) SchemaPath(path ...string) *CustomizableSchema {
+	sch := MustSchemaPath(s.GetSchemaMap(), path...)
+	return &CustomizableSchema{Schema: sch, path: path, context: s.context}
```

Comment:
> Don't we need to update the set of schemas in the context so that we inspect all intermediary schemas when verifying that there are no lists on this schema path?
Not really here, we need to do it for the path leading to the field that is set to be conflicted with the current field. This `SchemaPath` is not for that, this is help us to navigate to the field that we would like to customize.

Reply:
Ok, I understand. My main idea was that this implementation doesn't need to distinguish between potential parent fields in terms of the context and the fields specified in the `path` parameter. I suppose the main consequence is that you would not be able to catch misconfiguration where someone sets `ConflictsWith` inappropriately in the base structure (i.e. part of the specified `path` is a non-unary list). The provider framework should catch this in validation though, so I think that's OK.